stdlib/public/stubs/SwiftNativeNSXXXBase.mm.gyb

```diff
@@ -105,6 +105,22 @@ swift_stdlib_NSObject_isEqual(NSObject *NS_RELEASES_ARGUMENT lhs,
   return Result;
 }
 
+// _direct type metadata for Swift._NSContiguousString
+SWIFT_RUNTIME_STDLIB_INTERFACE
+extern "C" ClassMetadata _TMCs19_NSContiguousString;
+
+extern "C" {
+  struct _SwiftStackAllocatedNSContiguousString {
+    // Heap object header.
+    struct HeapObject header;
+
+    // _StringCore.
+    void *baseAddress;
+    uintptr_t countAndFlags;
+    void *owner;
+  };
```

Comment:
The allocation could probably be done on the swift side you would still call a library function to do the work (i.e. call  swift_stdlib_hashNSString(string: OpaquePointer, isASCII: Bool) but I think this might be a good way to cut down on some of the code duplication in allocating on the stack.